data-model/src/main/java/org/hypertrace/core/datamodel/shared/trace/StructuredTraceBuilder.java

```diff
@@ -191,6 +191,7 @@ private StructuredTrace build() {
     builder.setEventEdgeList(new ArrayList<>());
     builder.setEntityEdgeList(new ArrayList<>());
     builder.setEntityEventEdgeList(new ArrayList<>());
+    builder.setResourceList(this.resourceList);
```

Comment:
Being public method `buildStructuredTrace`, in this case, we may have resource_list, but resource_index can be default in Event. If, we are not using this method outside, except `buildStructuredTraceFromRawSpans`, can we consider making it `private`?

Reply:
Can you rephrase? I'm not really sure what you're asking. This method, `build` is private. `resourceList` part of StructuredTrace and is generated, along with the builder we're calling here.

Reply:
I was referring to `buildStructuredTrace` being public. So, in the below-calling order, we will have resurce_index as `-1` for the Event. 
```
StructuredTraceBuilder builder = new StructuredTraceBuilder(..., resourceList);
builder.buildStructuredTrace()
```
But, it seems to me that `buildStructuredTrace` is not used outside of this class. So, was suggesting, if we can make that `private`.

Reply:
I think this gets at how it's meant to be used. The existing design of this feels problematic for a number of reasons, but I was trying to work within it. The structured trace builder takes in events but does not mutate them - given that (which seems right to me), your scenario would work correctly. The problem in your scenario would be if we used the constructor with events that haven't been associated with resources yet. We'd have the same problem in other misconfigurations.

As that's the only non-private method on the builder, making it private is equivalent to making the whole class private and just exposing a public static util method. Is that what you're suggesting? I don't even think we should have this class at all to be honest, so I'm fine with that to start limiting it's usages.

Reply:
I agree with you about the need for this class. Yeah, I was thinking of exposing only the util method -`buildStructuredTraceFromRawSpans` which is only getting used in the HT side and restricting its usage. But, overall I am fine with model design.